### PR TITLE
feat: saving and logging

### PR DESCRIPTION
**Pull request by raodz**
## Description
Added some saving: training dir, model, config, plots.

## Testing
I checked - everything works besides model plot, there's a problem with graphviz

## Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Tests update
- [ ] Documentation update
- [ ] Changes that do not directly affect the source code or tests

## Checklist
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Additional Comments
in progress

### CODE REVIEW

File: scripts/train.py [hunk 1]
```diff
@@ -1,15 +1,30 @@
...
 import json
 from dataclasses import asdict
 from src.config import BoldOrNotConfig
+from src.logging import setup_logging
 from src.model_training import train_model
+from src.plot import plot_training_curves
+
+project_path = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+current_date = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+current_training = f"training{current_date}"
```
Comment (skrzypczykt) [inline scripts/train.py]: move to config,
`exp_name = config["exp_name"]`
`current_training_name = f"{exp_name }_{current_date}"`

File: scripts/train.py [hunk 2]
```diff
@@ -1,15 +1,30 @@
...
+training_dir = os.path.join(project_path, "trainings", current_training)
+os.makedirs(training_dir, exist_ok=True)
+shutil.copy("../config.yaml", os.path.join(training_dir, "config.yaml"))
```
Comment (skrzypczykt) [inline scripts/train.py]: make the config path a variable

Comment (skrzypczykt) [inline scripts/train.py]: make it a function and move to utils

File: src/model_training.py [hunk 3]
```diff
@@ -1,53 +1,110 @@
...
+import logging
+
+# from tensorflow.keras.utils import plot_model
 from src.data import BaldDataset
 from src.model import BaldOrNotModel
 from src.config import BoldOrNotConfig
 
 
-def train_model(config: BoldOrNotConfig):
+def train_model(config: BoldOrNotConfig, training_dir: str):
```
Comment (skrzypczykt) [inline src/model_training.py]: `training_dir` -> 'results_dir' albo 'output_dir`

File: src/model_training.py [hunk 4]
```diff
@@ -1,53 +1,110 @@
...
+    # Initialize model
     model = BaldOrNotModel(
-                dense_units=config.model_params.dense_units,
-                freeze_backbone=config.model_params.freeze_backbone,
-                dropout_rate=config.model_params.dropout_rate
-            )
-    optimizer = tf.keras.optimizers.Adam(learning_rate=config.training_params.learning_rate)
+        dense_units=config.model_params.dense_units,
+        freeze_backbone=config.model_params.freeze_backbone,
+        dropout_rate=config.model_params.dropout_rate,
```
Comment (skrzypczykt) [inline src/model_training.py]: **config.model_params

File: src/model_training.py [hunk 5]
```diff
@@ -1,53 +1,110 @@
...
+    )
+    logging.info(
```
Comment (skrzypczykt) [inline src/model_training.py]: too much

File: src/model_training.py [hunk 6]
```diff
@@ -1,53 +1,110 @@
...
         epochs=config.training_params.epochs,
         validation_data=train_dataset,
-        callbacks=tf_callbacks
+        callbacks=tf_callbacks,
     )
+    logging.info("Model training completed")
 
-    return history
+    # Save model and plot
+    model_path = os.path.join(training_dir, "model.keras")
```
Comment (skrzypczykt) [inline src/model_training.py]: `"model.keras"` move to **consts**

File: src/plot.py [hunk 7]
```diff
@@ -78,3 +87,54 @@ def plot_proportions(
         )
 
     plt.show()
+
+
+def plot_training_curves(history, training_dir):
```
Comment (skrzypczykt) [inline src/plot.py]: type hints

File: src/plot.py [hunk 8]
```diff
@@ -78,3 +87,54 @@ def plot_proportions(
...
+    if "val_loss" in history.history:
+        plt.plot(history.history["val_loss"], label="Loss (validation)")
+
+    # Add title, legend, and axis labels
+    plt.title("Loss Curves")
+    plt.xlabel("Epoch")
+    plt.ylabel("Loss")
+    plt.legend()
+
+    # Save the plot as a PNG file
```
Comment (skrzypczykt) [inline src/plot.py]: Close in a function, do not repeat code for accuracy, make it work for other metrics too

File: src/plot.py [hunk 9]
```diff
@@ -78,3 +87,54 @@ def plot_proportions(
...
+        plt.xlabel("Epoch")
+        plt.ylabel("Accuracy")
+        plt.legend()
+
+        # Save the plot as a PNG file
+        acc_plot_path = os.path.join(training_dir, "accuracy_plot.png")
+        plt.savefig(acc_plot_path)
+        plt.close()
+
+    print(f"Training curves saved in directory: {training_dir}")
```
Comment (skrzypczykt) [inline src/plot.py]: print -> logger

**Review by skrzypczykt — APPROVED** [review]
